sympy/stats/stochastic_process_types.py

```diff
@@ -759,3 +766,212 @@ def limiting_distribution(self):
         eqs.append(sum(wi) - 1)
         soln = list(linsolve(eqs, wi))[0]
         return ImmutableMatrix([[sol for sol in soln]])
+
+
+class BernoulliProcess(DiscreteTimeStochasticProcess):
+    """
+    The Bernoulli process consists of repeated
+    independent Bernoulli trials with the same parameter `p`.
+    It's assumed that the probability `p` applies to every
+    trial and that the outcomes of each trial
+    are independent of all the rest. Therefore Bernoulli Processs
+    is Discrete State and Discrete Time Stochastic Process.
+
+    Parameters
+    ==========
+
+    sym: Symbol/string_types
+    p: Proability of getting success
+
+    Examples
+    ========
+
+    >>> from sympy.stats import BernoulliProcess, P, E
+    >>> B = BernoulliProcess("B", 0.7)
+
+    >>> B.state_space
+    FiniteSet(0, 1)
+
+    >>> (B.p).round(2)
+    0.70
+
+    >>> (B.probability_r_success(ntrials=5, rsuccess=5,evaluate=True)).round(2)
+    0.17
+
+    >>> (B.probability_of_rth_success(ntrial=5, rsuccess=3)).round(2)
+    0.19
+
+    >>> B.joint_distribution(B[1], B[2])
+    JointDistributionHandmade(Lambda((B[1], B[2]), Piecewise((0.7, Eq(B[1], 1)),
+    (0.3, Eq(B[1], 0)), (0, True))*Piecewise((0.7, Eq(B[2], 1)), (0.3, Eq(B[2], 0)),
+    (0, True))))
+
+    >>> E(2*B[1] + B[2]).round(2)
+    2.10
+
+    >>> P(B[1] < 1).round(2)
+    0.30
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Bernoulli_process
+    .. [2] https://mathcs.clarku.edu/~djoyce/ma217/bernoulli.pdf
+
+    """
+
+    index_set = S.Naturals0
+
+    @property
+    def p(self):
+        return self.args[3]
+
+    @property
+    def distribution(self):
+        return self.args[2]
+
+    def __new__(cls, sym, p):
+        if p > 1 or p < 0:
+            raise ValueError("Probability must be in between 0 and 1")
+        p = sympify(p)
+        sym = _symbol_converter(sym)
+        state_space = _set_converter([0,1])
```

Comment:
Isn't that too rigid? What if I want to pass heads and tails as the state pass. May be we can allow state_space as parameter in the constructor.

Reply:
Okay, I will change that.